src/prism-async.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import highlight from './highlight';
+
+const languages = [];
+
+export const registerLanguage = (_, language) => {
+  if(ReactHighlightAsync.refractor) {
+    return refractor.register(language);
+  } else {
+    languages.push(language);
+  }
+};
+
+export default class ReactHighlightAsync extends React.PureComponent {
+  static refractor = null;
+  static highlightInstance = (highlight(null, {}));
+  static refractorPromise = null;
+  
+  static _loadRefractor() {
+    ReactHighlightAsync.refractorPromise = import(/* webpackChunkName:"react-syntax-highlighter/refractor-import" */ 'refractor/core').then(({ default: refractor }) => {
```

Comment:
what is webpackChunkName for here? assuming allows custom names of dynamic import in webpack? wiill this have effect on people not using webpack?

Reply:
On users not having webpack it'll just be a comment that's ignored and the code split will have whatever name the bundler assigns to it.